R/aes_y_lims.R

```diff
@@ -0,0 +1,615 @@
+#' Given a ggplot object, update the y-axis scales
+#' plot - ggplot object. This is the plot whose scales we want to update.
+#' auto_scale - should the chart be auto-scaled or should we leave it as is
+#' @noRd
+update_scales <- function(plot, auto_scale, warn = F){
+
+  # check if we have a numeric y-variable
+  check_y_var <- check_for_y_var(plot)
+
+  # if we don't have a numeric y-variable then check whether the plot contains ageom_density or geom_histogram (GeomBar without a y-variable)
+  if (!check_y_var) {
+    layers <- plot$layers
+
+    for (j in seq_along(layers)){
+
+      # don't get y-aesthetic for geom_text objects
+      layer_type <- layers[[j]]$geom %>% class()
+
+      # if there isn't one but it is a density plot or a bar chart then it is either a density chart or a histogram so go ahead
+      if ("GeomDensity" %in% layer_type | "GeomBar" %in% layer_type) {
+
+        check_y_var <- T
+
+        break
+      }
+    }
+  }
+
+  # check if we want to include a second y-axis or not (check by looking at whether it has a non-zero width grob)
+  grobs <- ggplot2::ggplotGrob(plot)
+
+  test_sec_axis <- get_grob_width(grobs, grob_name = "axis-r")
+
+  sec_axis <- !(is.null(test_sec_axis) | test_sec_axis == 0)
+
+  # if the y-variable class is numeric, or the plot is a density or histogram, then update the chart scales
+  if(check_y_var){
+
+    suppressMessages({plot <- update_chart_scales(plot, auto_scale, sec_axis)})
+
+    # if the y-var class is NULL, send a warning message about the auto updating of chart scales
+  } else if(!check_y_var & warn == F){
+
+    warning("Could not identify the class of the y variable. This prevents the y-axis scales from being automatically updated to aesthetic values. To address this issue check that you have not edited the variable within your ggplot call (e.g. aes(y = 100 * var)). Instead make any changes before passing the dataset to ggplot (e.g. data %>% mutate(new_var = 100 * var) %>% ggplot(...)).")
+    warn <<- T

```

Comment:
Why does this assign to global namespace?

Reply:
Yeah this is a hacky way of not repeating the warning multiple times. But as it risks overrinding any variable called warn I should probably remove it.

Reply:
You could give it a name that is less likely to be used like `t61.warn` or something

Reply:
yep good thinking